crates/ink/ir/src/ir/item_impl/mod.rs

```diff
@@ -373,3 +379,35 @@ impl ItemImpl {
         &self.items
     }
 }
+
+/// Check if the the conditional compilation expression
+/// for any `cfg` attributes is satisfied.
+///
+/// # Example
+/// If we have an annotated item (e.g. message)
+/// ```ignore
+/// #[ink(message)]
+/// #[cfg(feature = "foo")]
+/// #[cfg(feature = "baz")]
+/// pub fn get(&self) -> String {
+///     self.value.clone()
+/// }
+/// ```
+/// The function would would iterate over the `cfg` attributes
+/// and return `true` if any of the feature flags are set
+pub fn is_code_span_enabled(attrs: &[Attribute]) -> bool {
+    let cfg_attrs = attrs
+        .iter()
+        .filter(|a| a.path.is_ident("cfg"))
+        .map(|a| &a.tokens);
+    if cfg_attrs.clone().count() == 0 {
+        return true
+    }
+    for attr_tokens in cfg_attrs {
+        let _s = attr_tokens.to_token_stream();
+        if cfg!(_s) {
```

Comment:
Does this even work with the token stream? `cfg!` is a compile time macro.

I've tried it and it doesn't appear to work. Even with the literal `cfg!(feature = "foo")` it doesn't work for some reason. `cfg!(unix)` literal does work.

I think we need to rethink the approach here, maybe see whether we can carry those attributes through to the generated code (I know it is everywhere)